velox/substrait/TypeUtils.cpp

```diff
@@ -17,45 +17,87 @@
 #include "velox/type/Type.h"
 
 namespace facebook::velox::substrait {
-
-bool isString(const TypePtr& type) {
-  switch (type->kind()) {
-    case TypeKind::VARCHAR:
-    case TypeKind::VARBINARY:
-      return true;
-    default:
-      break;
+namespace {
+std::vector<std::string> getRowTypesFromCompoundName(
+    const std::string& compoundName) {
+  // CompoundName is like ROW<BIGINT,DOUBLE>
+  // or ROW<BIGINT,ROW<DOUBLE,BIGINT>,ROW<DOUBLE,BIGINT>>
+  // the position of then delimiter is where the number of leftAngleBracket
+  // equals rightAngleBracket need to split.
+  std::vector<std::string> types;
+  std::vector<int> angleBracketNumEqualPos;
+  auto leftAngleBracketPos = compoundName.find("<");
+  auto rightAngleBracketPos = compoundName.rfind(">");
+  auto typesName = compoundName.substr(
+      leftAngleBracketPos + 1, rightAngleBracketPos - leftAngleBracketPos - 1);
+  int leftAngleBracketNum = 0;
+  int rightAngleBracketNum = 0;
+  for (auto index = 0; index < typesName.length(); index++) {
+    if (typesName[index] == '<') {
+      leftAngleBracketNum++;
+    }
+    if (typesName[index] == '>') {
+      rightAngleBracketNum++;
+    }
+    if (typesName[index] == ',' &&
+        rightAngleBracketNum == leftAngleBracketNum) {
+      angleBracketNumEqualPos.push_back(index);
+    }
+  }
+  int startPos = 0;
+  for (auto delimeterPos : angleBracketNumEqualPos) {
+    types.emplace_back(typesName.substr(startPos, delimeterPos - startPos));
+    startPos = delimeterPos + 1;
   }
-  return false;
+  types.emplace_back(typesName.substr(startPos, typesName.length() - startPos));
+  return types;
 }
 
-int64_t bytesOfType(const TypePtr& type) {
-  auto typeKind = type->kind();
-  switch (typeKind) {
-    case TypeKind::INTEGER:
-      return 4;
-    case TypeKind::BIGINT:
-    case TypeKind::REAL:
-    case TypeKind::DOUBLE:
-      return 8;
-    default:
-      VELOX_NYI("Returning bytes of Type not supported for type {}.", typeKind);
+std::string getNameBeforeDelimiter(
```

Comment:
probably not a big deal, but maybe return a string_view to avoid the copy?

Reply:
Made the change, but still need to copy because the results are being passed to functions that only take std::string: mapNameToTypeKind and toVeloxType.